designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala

```diff
@@ -336,18 +337,36 @@ class UIProcessValidator(
   private def formatErrors(errors: NonEmptyList[ProcessCompilationError]): ValidationResult = {
     val globalErrors     = errors.filter(_.isInstanceOf[ScenarioGraphLevelError])
     val propertiesErrors = errors.filter(_.isInstanceOf[ScenarioPropertiesError])
-    val nodeErrors = errors.filter { e =>
+    val otherErrors = errors.filter { e =>
       !globalErrors.contains(e) && !propertiesErrors.contains(e)
     }
+    val (nodeErrors, unclassifiedErrors) = otherErrors.partition(_.nodeIds.nonEmpty)
+    val invalidNodes =
+      nodeErrors
+        .flatMap(error => error.nodeIds.map(nodeId => nodeId -> PrettyValidationErrors.formatErrorMessage(error)))
+        .toGroupedMap
+
+    // There shouldn't be unclassified errors, but the current design of ProcessCompilationError allows for such errors
+    // Try to get rid of them and make the context of the error as accurate as possible
```

Comment:
1. Can you add a TODO here?
2. Can you add a comment that the intention is to present these FatalUnknownErrors as node errors if they have a nodeId and as global errors if not?